### ISSUE TITLE
Movies and TV shows played via widgets appear to bypass Plex 

### ISSUE THREAD

**Issue by CraftyClown** [croneter/PlexKodiConnect#396]
Hello mate,

as mentioned in the PKC thread on the Plex forums, I have a strange issue where films and tv shows when selected via the 'recently added' or 'on deck' plugin play fine, however appear to bypass Plex. That is to say that when they are playing neither the Plex website or Tautulli show any activity. 

I've attached a log for you to take a gander at.

Cheers

[kodi.log](https://github.com/croneter/PlexKodiConnect/files/1692363/kodi.log)

**Comment by CraftyClown** [comment]
So I've just updated to beta 2.0 and the issue still exists.

I decided to do some more testing with logging enabled to help you troubleshoot. The findings are quite interesting so hopefully the log will also give you a bit of a lead on what's going on here.

What I've discovered is is it's playback via the widgets that breaks the connection to Plex. I play a movie from either the deck or recently added widget and they will play back ok, however they do not show up on the Plex now playing web status or in Tautulli. Once I have tried playing the film from on deck or recently added it will exhibit the same behaviour if played from the main movie library. 

To confirm the movies on deck and recently added weren't already exhibiting this behaviour, I decided to play a film (Batman begins) first from the movie library where it played fine and appeared in both Tautulli and the now playing status. I then tried it from the recently added list and once it again it was not showing in now playing or Tautulli. I went back to the main library and played it for a third time and this time it was not showing up in now playing or Tautulli.

Here's another weird thing; On deck for TV shows is suddenly working again, but recently added for TV shows has completely disappeared! according to my skin settings it should be showing and I tried removing it and adding it again, but the widget just doesn't show up!

after a full scan of the database PKC reports errors and advises checking the log.

[kodi.log](https://github.com/croneter/PlexKodiConnect/files/1697350/kodi.log)

**Comment by croneter** [comment]
Could you update to 2.0.1 and then try again? I don't see anything suspicious in the logs. Please also note here which videos you started from where and when it wasn't behaving as expected

**Comment by CraftyClown** [comment]
I tried 2.01 last night but the behaviour is the same. The recently added TV widget has reappeared though, so I'm guessing that was a fix you made?

One strange factor here is that one of the widgets works fine. TV on deck behaves exactly as it should, it is only TV recently added and Movies on deck and recently added that seem to bypass Plex.

**Comment by croneter** [comment]
Try 2.0.2  ;-)

Am 07.02.2018 6:33 nachm. schrieb "CraftyClown" <notifications@github.com>:

I tried 2.01 last night but the behaviour is the same. The recently added
TV widget has reappeared though, so I'm guessing that was a fix you made?

One strange factor here is that one of the widgets works fine. TV on deck
behaves exactly as it should, it is only TV recently added and Movies on
deck and recently added that seem to bypass Plex.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/croneter/PlexKodiConnect/issues/396#issuecomment-363847211>,
or mute the thread

**Comment by CraftyClown** [comment]
2.0.2 fixes it perfectly, you absolute legend :D

**Comment by CraftyClown** [comment]
Crap! I spoke a little too soon :(

So it's mostly working, however any new content that appears in the widgets behaves as before with nothing showing in Plex status's now playing section or Tautullia.

So close :D

**Comment by croneter** [comment]
New as in existing content now showing up as half watched in the widgets?
Or new as in new Plex content? Debug log please ;-)


Am 08.02.2018 2:04 vorm. schrieb "CraftyClown" <notifications@github.com>:

**Comment by CraftyClown** [comment]
Hello mate, so we are talking about any new content that was added to plex and showed up in the recently added widgets after the update to 2.0.2. Everything prior to this works as expected.

As before everything works fine from the main library.

In the generated log I have first played an episode of a TV show that was added after the 2.0.1 update, 'The Tunnel S2 E1' from the recently added widget. It did not show up in Tautulli. I then played the same episode from the main library and this time it down show up in Tautulli.
Straight after this I played a TV episode from the recently added widget that was available prior to 2.0.2 'Flash S4 E13' and that immediately showed up in Tautulli as expected.


[kodi.log](https://github.com/croneter/PlexKodiConnect/files/1706885/kodi.log)

**Comment by croneter** [comment]
Looks like PKC somehow messed up your Plex users:
```
PLEX.plexbmchelper.subscribers: Sent server notification with parameters: {'hasMDE': '1', 'X-Plex-Provider-Version': '2.0.2', 'X-Plex-Platform': 'Windows', 'playQueueItemID': '73673', 'ratingKey': '108920', 'key': '/library/metadata/108920', 'duration': 2504294, 'X-Plex-Device-Name': u'PlexKodiConnect', 'X-Plex-Token': u'redacted', 'X-Plex-Model': 'unknown', 'X-Plex-Platform-Version': 'unknown', 'X-Plex-Client-Identifier': u'a97da020-ab5b-4f04-83a9-9a067577acd6', 'X-Plex-Device': 'Windows', 'state': 'playing', 'X-Plex-Product': 'PlexKodiConnect', 'time': 13070, 'X-Plex-Version': '2.0.2'} to https://192.168.1.3:32400/:/timeline
PLEX.librarysync: Our username Rich_Kodi, userid 17177815 did not match the session username CraftyClown with userid 1
```
Make sure that you're using the same Plex user in PKC as well as Tautulli. Does rebooting Kodi help? Try also to re-select your PMS.

**Comment by CraftyClown** [comment]
Ahhh, interesting.

So I have an admin user 'CraftyClown' and a managed user called 'Rich_Kodi' that I use just so I don't need to share all my libraries with Kodi.

When I log in to PKC under the Plex.tv tab it always says logged in as CraftyClown despite being logged in under connections as Rich_Kodi. Is this the problem? 

Does this potentially mean only my main admin account will work properly with PKC?

**Comment by CraftyClown** [comment]
So I decided to do a full reset of Kodi and the database... which made things worse as once again only TV on deck worked.

I then changed the user back to CraftyClown and did just a reset of the database, however TV on deck is still the only properly working widget. 

Urgggggh :(

**Comment by croneter** [comment]
Well that's weird! Please share debug logs, otherwise it'll be hard for me
to pinpoint the root cause


Am 08.02.2018 6:33 nachm. schrieb "CraftyClown" <notifications@github.com>:

So I decided to do a full reset of Kodi and the database... which made
things worse as once again only TV on deck worked.

I then changed the user back to CraftyClown and did just a reset of the
database, however TV on deck is still the only properly working widget.

Urgggggh :(

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/croneter/PlexKodiConnect/issues/396#issuecomment-364188150>,
or mute the thread

**Comment by CraftyClown** [comment]
It really is weird!

So this time I started Kodi up and first played an episode of Flash from TV on deck, which worked fine. I then tried playing the same episode from recently added and as expected Plex does not recognise it is being played. I then tried the movie Arrival from Movies on deck and The Cloverfield Paradox from Movies recently added and once more no recognition from Plex that they are playing. Finally I tried playing Cloverfield Paradox once more from the Movie library and it was recognised in Plex now playing / Tautulli etc.

Really hope this helps you troubleshoot the issue because I am completely baffled! :D 

[kodi.log](https://github.com/croneter/PlexKodiConnect/files/1709115/kodi.log)

**Comment by CraftyClown** [comment]
So I decided to uninstall PKC yesteday, remove all libraries and use Kodi's library cleaning tools before reinstalling PKC and adding back the widgets.

Sadly this made no difference whatsoever. TV on deck works fine, however the other three main widgets do not seem to register in Plex.

The thing I don't understand is why no one else seems to be reporting this as an issue?

**Comment by CraftyClown** [comment]
Hello mate, don't suppose you've made any headway with this one?

**Comment by croneter** [comment]
Does 2.0.3 help? I cannot reproduce your issues...

**Comment by CraftyClown** [comment]
I've tried 2.0.3 but sadly still the same behaviour. I tried a different skin as well, just to see if that would make a difference but no dice.

Is there anything I can do to help troubleshoot this? I'll happily try anything to get this working properly.

**Comment by CraftyClown** [comment]
Do you think there's any value to a clean install of Kodi? Or is that overkill?

**Comment by croneter** [comment]
I don't think that helps. You could try to delete the folder video.plexkodiconnect in the user data folder and then there in the addon data folder (when Kodi is not running). Also delete everything in Database except the "addon xxx" file.

Thats pretty much the same as a fresh install

**Comment by croneter** [comment]
Found the problem thanks to your detailed description but maaaaaan it gives me headaches :grimacing:

**Comment by CraftyClown** [comment]
Great that you've sussed it, but sorry to hear it's a nightmare!

Weirdly I got everything working last night by doing a full uninstall of PKC and then trying to install a much older beta. This didn't actually help and resulted in PKC throwing scanning errors and asking for a Kodi restart. I did another full uninstall, put the latest beta back on and what do you know? It's all working properly!

I haven't tried adding any new content however and I seem to recall that last time that was when the problems started. I will test that now and see if the problem returns.

So in a nutshell what is the problem as you see it? Is it something I would even understand? lol

**Comment by croneter** [comment]
Trust me, you don't want to know...

**Comment by CraftyClown** [comment]
lol, fair enough :)

Is there any value to me tinkering around any further? Or is this just something that will keep happening until you've had a chance to undertake this horrible mission?

I'm still surprised no one else seems to be having this problem.

**Comment by croneter** [comment]
No you'll just have to wait. It would affect others using Addon Paths eventually

**Comment by CraftyClown** [comment]
Ah cool. So at least I know I'm not going crazy :)

One last question, are you saying it's related to add on paths? I'm happy to temporarily stay on default mode if it's going to be a workaround.

**Comment by croneter** [comment]
That is the default mode. Direct paths (more difficult setup) shouldn't be affected yes

**Comment by CraftyClown** [comment]
Ermm, well sorry to confuse things once again but I've always had direct paths set up, so it definitely is affecting them.

**Comment by croneter** [comment]
Should be fixed with new PKC version 2.0.4. Let me know here if this should not be the case.

**Comment by pannal** [comment]
I'm having a somewhat similar issue, current 2.0.11 beta on osmc, direct paths enabled. 

Starting to watch an episode from an on deck widget doesn't reflect any progress after stopping mid-view. Playing it again afterwards starts the file from the beginning, no resume state available. 

Is this expected behavior with direct paths?

**Comment by croneter** [comment]
Thanks for your report @pannal 😉

**Comment by pannal** [comment]
OK, no. Even after a resync the progress isn't reflected in the TV shows on deck widget. It always is, on the PMS side, though.

What do you need from me in order to investigate this?

**Comment by croneter** [comment]
Kodi DEBUG Logs as described in the wiki please

**Comment by pannal** [comment]
[kodi.log](https://github.com/croneter/PlexKodiConnect/files/1834209/kodi.log)

Weirdly after enabling DEBUG and restarting Kodi (which I have done a couple of times before), it registers progress.

Those problems exist:
- it doesn't, visually. That an episode is in-progress is only visible by playing the item and waiting for the context menu to appear (or not)
- I skipped to minute 40 of the 44min runtime episode, then stopped playback. After waiting a bit and going up and down in the side menu, the episode received a "watched"-marker, and after hitting play, the context menu popped up asking me whether to continue from minute 40 or not. Either the watched marker is wrong and/or the watched state is determined incorrectly/inconsistently. Also I'd expect it to switch over to the next unwatched episode.

**Comment by pannal** [comment]
While browsing through the logs, I noticed the watched-state change:
`DEBUG: DoWork - Marking video item plugin://plugin.video.plexkodiconnect.tvshows/?plex_id=159830&plex_type=episode&mode=play as watched`

And about 3 seconds later the stopped event via WebSocket with a viewOffset:
`DEBUG: PLEX.websocket_client: PMS_Websocket: Received message from PMS server: {u'PlaySessionStateNotification': [{u'viewOffset': 1831029, u'playQueueItemID': 118271, u'url': u'', u'ratingKey': u'159830', u'state': u'stopped', u'sessionKey': u'86', u'key': u'/library/metadata/159830', u'guid': u''}], u'type': u'playing', u'size': 1}`

That might explain my second bulletpoint from the post above.

**Comment by croneter** [comment]
Thanks for the log. I do not see anything suspicious on first glance, although I don't see the 40 out of 44min, but this:
```
PLEX.player: Playback progress 0.689227369911 (1831.029 of 2656.64 seconds)
```
Which is 30min out of 44 - the item should NOT get marked as completely watched then (not above 90% completion). 

The `DEBUG: PLEX.websocket_client: PMS_Websocket: Received message`...  you mentioned is registered but (correctly) ignored. 

Having said that, I see that you're using Titan. I had issues with Titan in the past, esp. on potentially weak hardware. Refreshing of library items and watched state seemed random. Could you please try another skin?

**Comment by croneter** [comment]
I don't understand your first bullet
> * it doesn't, visually. That an episode is in-progress is only visible by playing the item and waiting for the context menu to appear (or not)

**Comment by pannal** [comment]
I updated the bullet just before you posted.

I'd expect a percentage shown on the item that is in-progress - nothing is shown instead. The only way to see if the item is in-progress  is by playing it, then the context menu appears asking me whether I want to resume or start from the beginning.


I'll try another skin using another profile and a fresh database. Which one do you currently use for your PKC?

**Comment by croneter** [comment]
Closing again as this is most probably related to the BETA version of the titan skin
